### PR TITLE
refactor:  Setup of Express (OOps based)(#39521)

### PR DESCRIPTION
**Pull request by ajayrose25**

### CODE REVIEW

File: src/.env [hunk 1]
```diff
@@ -0,0 +1,2 @@
+PORT=9000
```
Comment (VisaaliSuccessive) [inline src/.env]: move env out of src folder

File: src/Server.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import * as express from 'express';
+import Iconfig from './config/iconfig';
+
+class Server {
+    private app: express.Express;
+    constructor(private config: Iconfig) {
+        this.app = express();
+    }
+    bootstrap() {
+        console.log('inside bootstrap');
```
Comment (VisaaliSuccessive) [inline src/Server.ts]: remove the unwanted console log

File: src/Server.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        this.setupRoutes();
+        return this;
+    }
+    run = () => {
+        const { app, config: { port } } = this;
+        app.listen(port, (err) => {
+            if (err) {
+                throw err;
+            }
+            console.log('app is running successfully');
```
Comment (VisaaliSuccessive) [inline src/Server.ts]: fetch the env currently, you are working on and print the port on which this app is running through environment variables.

File: src/config/configuration.ts [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import { config } from 'dotenv';
+import Iconfig from './iconfig';
+config();
+
+const configuration: Iconfig = Object.freeze( {
+    port: process.env.PORT,
```
Comment (VisaaliSuccessive) [inline src/config/configuration.ts:6]: did you check whether this env is available here? As the .env file is not located at its right position

File: src/.env (hunk 1 above)
Comment (ajayrose25) [reply src/.env]: Done

File: src/Server.ts (hunk 2 above)
Comment (ajayrose25) [reply src/Server.ts]: Done

File: src/Server.ts (hunk 3 above)
Comment (ajayrose25) [reply src/Server.ts]: Done

File: src/config/configuration.ts (hunk 4 above)
Comment (ajayrose25) [reply src/config/configuration.ts:6]: Done

File: src/Server.ts [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+        this.setupRoutes();
+        return this;
+    }
+    run = () => {
+        const { app, config: { port } } = this;
+        app.listen(port, (err) => {
+            if (err) {
+                throw err;
+            }
+            console.log('app is running successfully on' , {port});
```
Comment (VisaaliSuccessive) [inline src/Server.ts]: add the current node env also as commented earlier

File: src/.env (hunk 1 above)
Comment (VisaaliSuccessive) [reply src/.env]: add env to gitignore also

File: src/Server.ts (hunk 5 above)
Comment (ajayrose25) [reply src/Server.ts]: Done

File: .env [hunk 6]
```diff
@@ -0,0 +1,2 @@
+PORT=9000
```
Comment (VisaaliSuccessive) [inline .env]: remove env file from git

Comment (ajayrose25) [reply .env]: Done

**Pull request merged**